src/control/server/telemetry.go

```diff
@@ -19,23 +19,32 @@ import (
 	"github.com/daos-stack/daos/src/control/logging"
 )
 
-func regPromEngineSources(ctx context.Context, log logging.Logger, engines []*EngineInstance) error {
+func regPromEngineSources(ctx context.Context, log logging.Logger, engines []*EngineInstance) ([]func(), error) {
```

Comment:
In hindsight, I'm not sure how useful those function signature aliases are... I think the complex ones like `systemJoinFn` and `onInstanceExitFn` make sense, but the others are questionable. My thinking at the time when I added them was that it might be better to have one place to change the definition a callback, especially because we were just starting to move to use them more often and they might need to evolve, but I think now the simple ones might just be pointless ceremony.